core/shared/src/main/scala/zio/prelude/fx/ZPure.scala

```diff
@@ -554,9 +586,16 @@ sealed trait ZPure[-S1, +S2, -R, +E, +A] { self =>
           a = updated._2
           val nextInstr = stack.pop()
           if (nextInstr eq null) curZPure = null else curZPure = nextInstr(a)
+        case Tags.Log     =>
+          val zPure     = curZPure.asInstanceOf[Log[Any, Any]]
+          log = log :+ zPure.log
```

Comment:
I would use a mutable array builder here.

Reply:
If we change this to array builder, then "get log" needs to do a copy, of course.

Reply:
Good idea!